tests/store/test_s3_artifact_repo.py

```diff
@@ -0,0 +1,56 @@
+import os
+import unittest
+
+import boto3
+from moto import mock_s3
+
+from mlflow.store.artifact_repo import ArtifactRepository, S3ArtifactRepository
+from mlflow.utils.file_utils import TempDir
+from tests.helper_functions import random_int
+
+
+class TestS3ArtifactRepo(unittest.TestCase):
+    @mock_s3
+    def test_basic_functions(self):
```

Comment:
I think it's hard to do that because you need to add some artifacts in order to list them and such. Seems easier to have one test for basic stuff and then other ones for any special cases we might encounter later.